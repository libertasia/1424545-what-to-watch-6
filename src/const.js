const DEFAULT_GENRE = `All genres`;

const MAX_GENRES_COUNT = 9;

const DEFAULT_VISIBLE_FILMS_COUNT = 8;

const NUMBER_OF_SECONDS_IN_HOUR = 3600;

const NUMBER_OF_MINUTES_IN_HOUR = 60;

const PLAYER_TOGGLER_WIDTH = 17;

const AuthorizationStatus = {
  INIT: `INIT`,
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
  AUTH_ERROR: `AUTH_ERROR`,
};

const Rating = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  AWESOME: 10,
};

const RatingLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

const AppRoute = {
  ADD_REVIEW: `/films/:id/review`,
  FILM: `/films/:id`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  PLAYER: `/player/:id`,
  ROOT: `/`,
};

const APIRoute = {
  ERROR_404: `/404`,
  FAVORITE_FILMS: `/favorite`,
  FILMS: `/films`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  PROMO_FILM: `/films/promo`,
};

const TabTypes = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

const MouseEvent = {
  MOUSE_MOVE: `mousemove`,
  MOUSE_UP: `mouseup`,
};

export {DEFAULT_GENRE, MAX_GENRES_COUNT, DEFAULT_VISIBLE_FILMS_COUNT, AuthorizationStatus, Rating, RatingLevel, APIRoute, AppRoute, TabTypes, MouseEvent, NUMBER_OF_MINUTES_IN_HOUR, NUMBER_OF_SECONDS_IN_HOUR, PLAYER_TOGGLER_WIDTH};
