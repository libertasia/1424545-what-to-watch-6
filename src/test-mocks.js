const mockFilms = [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    posterImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    previewImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    backgroundImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    backgroundColor: `#ffffff`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    released: 2014,
    isFavorite: true
  },
  {
    id: 2,
    name: `Stand By Me`,
    posterImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    previewImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    backgroundImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    backgroundColor: `#ffffff`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Rob Reiner's adaptation of Stephen King's novella The Body is a stirring, touching adventure film which knows the real world is exciting and scary enough just as it is. It's also a coming-of-age movie which celebrates the intensity of childhood friendship, while gently mourning the transience of such bonds. Which is why, unlike its central character, it'll never get old.`,
    rating: 7,
    scoresCount: 180,
    director: `Rob Reiner`,
    starring: [`River Phoenix`, `Wil Wheaton`, `Corey Feldman`, `Jerry O'Connell`],
    runTime: 89,
    genre: `Drama`,
    released: 1986,
    isFavorite: true
  },
  {
    id: 3,
    name: `Amelie`,
    posterImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    previewImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    backgroundImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
    backgroundColor: `#ffffff`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Despite being caught in her imaginative world, Amelie, a young waitress, decides to help people find happiness. Her quest to spread joy leads her on a journey where she finds true love.`,
    rating: 9,
    scoresCount: 355,
    director: `Jean-Pierre Jeunet`,
    starring: [`Audrey Tautou`, `Mathieu Kassovitz`, `Jamel Debbouze`],
    runTime: 120,
    genre: `Drama`,
    released: 2001,
    isFavorite: true
  },
];

const mockFilm = {
  id: 6,
  name: `The Exorcist`,
  posterImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
  previewImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
  backgroundImage: `https://i.picsum.photos/id/604/248/152.jpg?hmac=Yu4rnPqlL8pPa8b7cdPSosfFApQXtQmy56BW52elqto`,
  backgroundColor: `#ffffff`,
  videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  description: `An actress notices dangerous changes in the behavior and physical make-up of her 12-year-old daughter. Meanwhile, a young priest begins to doubt his faith while dealing with his mother's sickness.`,
  rating: 8.8,
  scoresCount: 174,
  director: `William Friedkin`,
  starring: [`Linda Blair`, `Ellen Burstyn`, `Jason Miller`, `Max von Sydow`],
  runTime: 128,
  genre: `Thriller`,
  released: 1973,
  isFavorite: true
};

const mockReviews = [
  {
    id: 1,
    user: {
      id: 17,
      name: `Emely`
    },
    rating: 9.8,
    comment: `This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.`,
    date: `2021-01-27T13:18:14.644Z`
  },
  {
    id: 2,
    user: {
      id: 13,
      name: `Zak`
    },
    rating: 8.4,
    comment: `This movie is perfect in all its categories: credits, sound track, production, casting, writing, photography, editing, acting, and direction.\nI was amazed with the freedom of the use of the camera. This movie will change the way movies are made. Slow-mo, stills, black and white, and color were all used to brilliant effect.`,
    date: `2021-01-27T13:18:14.644Z`
  },
  {
    id: 3,
    user: {
      id: 17,
      name: `Emely`
    },
    rating: 5.7,
    comment: `This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.`,
    date: `2021-02-06T13:18:14.644Z`
  }
];

export {mockFilms, mockFilm, mockReviews};
