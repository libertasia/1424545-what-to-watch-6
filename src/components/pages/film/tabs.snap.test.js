import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from 'history';
import {Tabs} from './tabs';
import {mockFilm, mockReviews} from '../../../test-mocks';
import {TabTypes} from '../../../const';

const mockStore = configureStore({});
const store = {
  FILMS: {
    film: mockFilm,
    reviews: mockReviews,
    activeTab: TabTypes.OVERVIEW,
  }
};
describe(`Tabs should render correctly`, () => {
  it(`Tabs should render correctly if active tab is Overview`, () => {
    const history = createMemoryHistory();
    const {container} = render(
        <Provider store={mockStore(store)}>
          <Router history={history}>
            <Tabs
              film={mockFilm}
              reviews={mockReviews}
              activeTab={TabTypes.OVERVIEW}
              setActiveTab={jest.fn()}
            />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  it(`Tabs should render correctly if active tab is Details`, () => {
    const history = createMemoryHistory();
    const {container} = render(
        <Provider store={mockStore(store)}>
          <Router history={history}>
            <Tabs
              film={mockFilm}
              reviews={mockReviews}
              activeTab={TabTypes.DETAILS}
              setActiveTab={jest.fn()}
            />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  it(`Tabs should render correctly if active tab is Reviews`, () => {
    const history = createMemoryHistory();
    const {container} = render(
        <Provider store={mockStore(store)}>
          <Router history={history}>
            <Tabs
              film={mockFilm}
              reviews={mockReviews}
              activeTab={TabTypes.REVIEWS}
              setActiveTab={jest.fn()}
            />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  // it(`Click on FavoriteButton works`, () => {
  //   const favoriteButtonClickHandler = jest.fn();
  //   favoriteButtonClickHandler.mockImplementation(
  //       () => (mockFilm.isFavorite = false)
  //   );

  //   render(
  //       <Provider store={mockStore(store)}>
  //         <FavoriteButton
  //           film = {mockFilm}
  //           onClick={favoriteButtonClickHandler}
  //         />
  //       </Provider>
  //   );

  //   expect(screen.getByText(/My list/i)).toBeInTheDocument();
  //   userEvent.click(screen.getByRole(`button`));
  //   expect(favoriteButtonClickHandler).toBeCalled();
  //   expect(mockFilm.isFavorite).toBe(false);
  // });
});
