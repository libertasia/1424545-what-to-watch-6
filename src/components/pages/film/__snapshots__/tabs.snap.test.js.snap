// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly Tabs should render correctly if active tab is Details 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item "
          data-id="overview"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item movie-nav__item--active"
          data-id="details"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item "
          data-id="reviews"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-card__text movie-card__row"
    >
      <div
        class="movie-card__text-col"
      >
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Director
          </strong>
          <span
            class="movie-card__details-value"
          >
            William Friedkin
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Starring
          </strong>
          <span
            class="movie-card__details-value"
          >
            Linda Blair,
            <br />
            Ellen Burstyn,
            <br />
            Jason Miller,
            <br />
            Max von Sydow,
            <br />
          </span>
        </p>
      </div>
      <div
        class="movie-card__text-col"
      >
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Run Time
          </strong>
          <span
            class="movie-card__details-value"
          >
            2h 8m
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Genre
          </strong>
          <span
            class="movie-card__details-value"
          >
            Thriller
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Released
          </strong>
          <span
            class="movie-card__details-value"
          >
            1973
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs should render correctly Tabs should render correctly if active tab is Overview 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item movie-nav__item--active"
          data-id="overview"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item "
          data-id="details"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item "
          data-id="reviews"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-rating"
    >
      <div
        class="movie-rating__score"
      >
        8.8
      </div>
      <p
        class="movie-rating__meta"
      >
        <span
          class="movie-rating__level"
        >
          Very good
        </span>
        <span
          class="movie-rating__count"
        >
          174
           ratings
        </span>
      </p>
    </div>
    <div
      class="movie-card__text"
    >
      <p>
        An actress notices dangerous changes in the behavior and physical make-up of her 12-year-old daughter. Meanwhile, a young priest begins to doubt his faith while dealing with his mother's sickness.
      </p>
      <p
        class="movie-card__director"
      >
        <strong>
          Director: 
          William Friedkin
        </strong>
      </p>
      <p
        class="movie-card__starring"
      >
        <strong>
          Starring: 
          Linda Blair, Ellen Burstyn, Jason Miller, Max von Sydow
           and other
        </strong>
      </p>
    </div>
  </div>
</div>
`;

exports[`Tabs should render correctly Tabs should render correctly if active tab is Reviews 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item "
          data-id="overview"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item "
          data-id="details"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item movie-nav__item--active"
          data-id="reviews"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-card__reviews movie-card__row"
    >
      <div
        class="movie-card__reviews-col"
      >
        <div
          class="review"
        >
          <blockquote
            class="review__quote"
          >
            <p
              class="review__text"
            >
              This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
            </p>
            <footer
              class="review__details"
            >
              <cite
                class="review__author"
              >
                Emely
              </cite>
              <time
                class="review__date"
                datetime="2021-01-27T13:18:14.644Z"
              >
                January 27, 2021
              </time>
            </footer>
          </blockquote>
          <div
            class="review__rating"
          >
            9.8
          </div>
        </div>
        <div
          class="review"
        >
          <blockquote
            class="review__quote"
          >
            <p
              class="review__text"
            >
              This movie is perfect in all its categories: credits, sound track, production, casting, writing, photography, editing, acting, and direction.
I was amazed with the freedom of the use of the camera. This movie will change the way movies are made. Slow-mo, stills, black and white, and color were all used to brilliant effect.
            </p>
            <footer
              class="review__details"
            >
              <cite
                class="review__author"
              >
                Zak
              </cite>
              <time
                class="review__date"
                datetime="2021-01-27T13:18:14.644Z"
              >
                January 27, 2021
              </time>
            </footer>
          </blockquote>
          <div
            class="review__rating"
          >
            8.4
          </div>
        </div>
        <div
          class="review"
        >
          <blockquote
            class="review__quote"
          >
            <p
              class="review__text"
            >
              This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.
            </p>
            <footer
              class="review__details"
            >
              <cite
                class="review__author"
              >
                Emely
              </cite>
              <time
                class="review__date"
                datetime="2021-02-06T13:18:14.644Z"
              >
                February 6, 2021
              </time>
            </footer>
          </blockquote>
          <div
            class="review__rating"
          >
            5.7
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
